### ISSUE TITLE
MCT configure doesn't work with Intel OneAPI

### ISSUE THREAD

**Issue by rljacob** [MCSclimate/MCT#67]
When trying to run configure with FC=ifx, it fails when checking for Fortran-C name mangling.

```
checking for dummy main to link with Fortran libraries... unknown
configure: error: in `/home/jacob/MCT':
configure: error: linking to Fortran libraries from C fails

```
Reported by @naromero77

**Comment by rljacob** [comment]
This discussion suggests its a known problem that was fixed in autoconf 2.70

https://community.intel.com/t5/Intel-Fortran-Compiler/ifx-2021-1-beta04-HPC-Toolkit-build-error-with-loopopt/td-p/1184181

But a PR with an autoconf 2.70 configure didn't fix it.

**Comment by rljacob** [comment]
Maybe its a problem with the special beta we are using:
```
[jacob@jlselogin2 MCT]$ ifx --version
ifx (IFORT) 2021.1 Beta 20201113
Copyright (C) 1985-2020 Intel Corporation. All rights reserved.
```
